### ISSUE TITLE
Consider smoothing out metrics

### ISSUE THREAD

**Issue by 2opremio** [weaveworks/scope#1279]
Metrics (e.g. CPU/memory consumption) are pretty spiky (probably due to our sampling). This is particularly obvious on the sparklines, due to their narrow vertical autoscaling.

![](http://g.recordit.co/3xIFyXGRO6.gif)

We should consider smoothing them out with a [moving average](https://en.wikipedia.org/wiki/Moving_average), whose width could be configured by the end user (or not) .

If we choose to do it, it should be done consistently everywhere (see #1278 ).

**Comment by foot** [comment]
I think spikeyness is by design! But that is a discussion we should have.

The other way to de-spike is to use the "real" max, rather than the 60s max, which will make all the spikes look tiny.

**Comment by 2opremio** [comment]
60s max? Is that for vertical scaling?

**Comment by foot** [comment]
Yep, atm we just use the max of the values in the sparkline. It shows you what's been happening in the last 15-60s (depending on how long you give it), nicely.

We recalculate the y-range (vertical scaling) based on the local max.
